baselines/dom.generated.d.ts

```diff
@@ -5267,9 +5298,13 @@ interface HTMLAnchorElement extends HTMLElement, HTMLHyperlinkElementUtils {
      */
     text: string;
     type: string;
-    addEventListener<K extends keyof HTMLElementEventMap>(type: K, listener: (this: HTMLAnchorElement, ev: HTMLElementEventMap[K]) => any, options?: boolean | AddEventListenerOptions): void;
+    addEventListener<K extends keyof DocumentAndElementEventHandlersEventMap>(type: K, listener: (this: HTMLAnchorElement, ev: DocumentAndElementEventHandlersEventMap[K]) => any, options?: boolean | AddEventListenerOptions): void;
+    addEventListener<K extends keyof ElementEventMap>(type: K, listener: (this: HTMLAnchorElement, ev: ElementEventMap[K]) => any, options?: boolean | AddEventListenerOptions): void;
```

Comment:
`HTMLElementEventMap` already extends `ElementEventMap`

Reply:
also i think the change here should be on `HTMLElementEventMap` to extend `DocumentAndElementEventHandlersEventMap` and `GlobalEventHandlersEventMap` instead of on every element.

Reply:
>HTMLElementEventMap already extends ElementEventMap

Yes, but that map is now empty and not generated because HTMLElement does not directly include any event handlers anymore.

Reply:
we should generate the map always instead of generating overloads.. the map can still be empty, but just extends other maps.